### PR TITLE
Adds Yes/No to ActiveModel::Type::Boolean options.

### PR DESCRIPTION
**Pull request by fabionl**
### Summary

My experience shows that end users tend to understand Yes/No pair better than True/False. This is a very common case in my projects when consuming CSV files provided by users, for example. As it seems to me that Yes/No are not ambiguous terms, I'm hoping ActiveModel could handle them without any major risks to existing codebases.

### Other Information

I looked at the documentation and I believe it doesn't need to be changed, but I'm glad to make further changes as seen fit.
I believe this is self containing? Happy to edit if I'm missing any information.

### CODE REVIEW

File: activemodel/lib/active_model/type/boolean.rb [hunk 1]
```diff
@@ -23,6 +23,8 @@ class Boolean < Value
         "FALSE", :FALSE,
         "off", :off,
         "OFF", :OFF,
+        "no", :no,
+        "NO", :NO,
```
Comment (bdewater) [inline activemodel/lib/active_model/type/boolean.rb:27]: Doesn't this introduce the Norway ('NO' is the ISO 3166 country code) problem YAML has? Also see https://hitchdev.com/strictyaml/why/implicit-typing-removed/

Comment (fabionl) [reply activemodel/lib/active_model/type/boolean.rb:27]: Ok, that's interesting. I wasn't aware of this type of issues. In my case, we explicitly use ActiveModel::Type::Boolean in some places when processing CSV files, for example.

I guess the question would be, is this YAML issue a reason not to let AciveModel::Type::Boolean cast 'No' as a valid boolean? Also, is there something I could do to help mitigate the issue and still have the ability to cast it?

**Comment by fabionl** [comment]
Any chance of receiving guidance from someone on this? I wonder what else I could do to move forward.

File: activemodel/test/cases/type/boolean_test.rb [hunk 2]
```diff
@@ -41,13 +45,17 @@ def test_type_cast_boolean
         assert_equal false, type.cast("FALSE")
         assert_equal false, type.cast("off")
         assert_equal false, type.cast("OFF")
+        assert_equal false, type.cast("no")
+        assert_equal false, type.cast("no")
```
Comment (websebdev) [inline activemodel/test/cases/type/boolean_test.rb:49]: I believe this should be in caps "NO".

Comment (fabionl) [reply activemodel/test/cases/type/boolean_test.rb]: Fixed. Thanks!

**Comment by DRBragg** [comment]
FWIW I'm pro this change!

**Comment by fabionl** [comment]
Is there an interest to move forward with this change? Am I still missing something? I'm willing to make any further changes that might help.

**Comment by rafaelfranca** [comment]
We prefer to not change this type in Rails since it can break existing applications. But feel free to define your own Boolean type and change the type registry to use your type by default.

**Pull request closed without merging**